src/libs/extra/metadata.liq

```diff
@@ -63,40 +30,42 @@ def file.cover.manager(
       "Extracting cover from #{string.quote(filename)}."
     )
     cover = metadata.cover(_metadata)
-    let (cover_file, mime) =
-      null.case(cover, {(null(), "")}, save_cover_to_file)
 
-    next_cover =
+    new_cover =
       if
-        cover == null()
-      then
-        log.important(
-          label=id,
-          "File #{string.quote(filename)} has no cover."
-        )
-        default
-      elsif
-        cover_file == null()
+        not null.defined(cover)
       then
-        log.important(
-          label=id,
-          "File #{string.quote(filename)} has unknown mime type #{
-            string.quote(mime)
-          }."
-        )
-        default
+        null()
       else
-        cover_file = null.get(cover_file)
-        log.important(
-          label=id,
-          "Cover for #{string.quote(filename)} saved to #{
-            string.quote(cover_file)
-          }."
-        )
-        cover_file
+        cover = null.get(cover)
+        extension = mime_types[cover.mime]
+        if
+          extension == ""
+        then
+          log.important(
+            label=id,
+            "File #{string.quote(filename)} has unknown mime type #{
+              string.quote(cover.mime)
+            }."
+          )
+          null()
+        else
+          cover_file = file.temp("#{id}_", ".#{extension}")
+          file.write(cover_file, data=cover)
+
+          log.important(
+            label=id,
+            "Cover for #{string.quote(filename)} saved to #{
+              string.quote(cover_file)
+            }."
+          )
+
+          request.create(temporary=true, cover_file)
+        end
       end
-    set_current_cover(next_cover)
+
+    current_cover_request := (new_cover ?? default)
```

Comment:
Does it make sense to return `null()`?

Reply:
We need to reset to `default` when the file has no cover, right?

Reply:
I'd like to point out that the previous if statement could return `default` instead of `null()`, so the whole if statement will assign a value to `current_cover_request`.

Reply:
It doesn't look like so. But if you find it more readable, we could return `default` from within the `if` statement:
```liquidsoap
    new_cover =
      if
        not null.defined(cover)
      then
        log.important(
          label=id,
          "File #{string.quote(filename)} has no cover."
        )
        null()
      else
        cover = null.get(cover)
        extension = mime_types[cover.mime]
        if
          extension == ""
        then
          log.important(
            label=id,
            "File #{string.quote(filename)} has unknown mime type #{
              string.quote(cover.mime)
            }."
          )
          null()
        else
          cover_file = file.temp("#{id}_", ".#{extension}")
          file.write(cover_file, data=cover)

          log.important(
            label=id,
            "Cover for #{string.quote(filename)} saved to #{
              string.quote(cover_file)
            }."
          )

          request.create(temporary=true, cover_file)
        end
      end
```

Reply:
Never mind, it's fine.